### DISCUSSION TITLE
Does anyone know of a grammar that uses github actions to deploy a github-hosted website serving the playground for that grammar?

### DISCUSSION THREAD

**Discussion by ahelwer** [tree-sitter/tree-sitter#1299] [Q&A] [answered]
Would be a neat way to easily demo the work. Will take a crack at it but thought I'd see whether someone has already figured out how to do all this.

**Comment by stsewd** [comment, marked as answer]
I have done this with https://github.com/stsewd/tree-sitter-rst/tree/master/docs, preview at https://stsewd.dev/tree-sitter-rst/

**Reply by ahelwer** [reply]
@stsewd you do this using submodules in your stsewd.github.io repo?

**Reply by stsewd** [reply]
@ahelwer no, you can host any github pages from any repo, it is served as `roourl/{project-name}`, you only need to activate it from your repo settings.

**Reply by ahlinc** [reply]
Another example can looked at https://github.com/ananthakumaran/tree-sitter-elixir and its deployed playground here: https://ananthakumaran.in/tree-sitter-elixir/

**Reply by ahelwer** [reply]
Great, got it working, thanks! https://tlaplus-community.github.io/tree-sitter-tlaplus/

**Reply by razzeee** [reply]
You got a stray </a> here https://github.com/tlaplus-community/tree-sitter-tlaplus/blob/playground/docs/index.html#L24

Anybody knows what I'm doing wrong? https://elm-tooling.github.io/tree-sitter-elm/
https://github.com/elm-tooling/tree-sitter-elm

The wasm is created by doing `npx tree-sitter build-wasm` right? Docker build should be fine too?

**Reply by ahelwer** [reply]
I just run `tree-sitter build-wasm`. If you look at your dev console on that site you see a lot of these:
```
Uncaught (in promise) RuntimeError: indirect call to null
    parse https://elm-tooling.github.io/tree-sitter-elm/js/tree-sitter.js:1
    handleCodeChange https://elm-tooling.github.io/tree-sitter-elm/js/playground.js:115
    it https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.45.0/codemirror.min.js:1
    jn https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.45.0/codemirror.min.js:1
    Gn https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.45.0/codemirror.min.js:1
    Gn https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.45.0/codemirror.min.js:1
    Gn https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.45.0/codemirror.min.js:1
    Gn https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.45.0/codemirror.min.js:1
    n https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.45.0/codemirror.min.js:1
    Cl.prototype[Xl]</< https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.45.0/codemirror.min.js:1
    handle https://elm-tooling.github.io/tree-sitter-elm/:83
tree-sitter.wasm:108197:1
```
Could be related to my current issue which involves the external scanner, see https://github.com/tree-sitter/tree-sitter/issues/1300

Does `tree-sitter playground` work fine locally? Also is your `emcc` version less than 2.0.17 (see [this issue](https://github.com/tree-sitter/tree-sitter/issues/1098))?

**Reply by razzeee** [reply]
I'm getting a not working version with `tree-sitter playground` too, thought nothing of that earlier.

I'm using docker, so I'm not sure which `emcc` verison get's used, but the image should be up to date, as I've resetup this laptop four days ago.

**Reply by ahelwer** [reply]
Yeah a too-new version of emcc will break it. You might need to install emcc 2.0.17 or earlier directly.

**Reply by razzeee** [reply]
Oh well, seems like the local playground now works fine.

**Reply by razzeee** [reply]
Got it to work, I've run `tree-sitter playground` and grabbed the `tree-sitter.js` from there and used that in my repo...

https://github.com/elm-tooling/tree-sitter-elm/commit/b72303375b29b0fb77021d777818c64bf1664f03

**Reply by ahelwer** [reply]
So you can pre-set the code window, anyone have some example code for how to pre-set the query window? And how you can check the query box by default?
